### PR TITLE
Unit tests for audio settings saveload

### PR DESCRIPTION
**Pull request by ikostan**
---
name: Default Pull Request Template
about: Suggesting changes to SkyLockAssault
title: ''
labels: ''
assignees: ''
---

## Description

What does this PR do? (e.g., "Fixes player jump physics in level 2" or "Adds
new enemy AI script")

## Related Issue

Closes #ISSUE_NUMBER (if applicable)

## Changes

- [x] List key changes here (e.g., "Updated Jump.gd to use Godot 4.4's new Tween
  system")
- [x] Any breaking changes? (e.g., "Deprecated old signal; migrate to new one")

## Testing

- [x] Ran the game in Godot v4.5 editor—describe what you tested (e.g., "Jump
  works on Win10 with 60 FPS")
- [x] Any new unit tests added? (Link to test scene if yes)
- [x] Screenshots/GIFs if UI-related: (Attach below)

## Checklist

- [x] Code follows Godot style guide (e.g., snake_case for variables)
- [x] No console errors in editor/output
- [x] Ready for review!

## Additional Notes

Anything else? (e.g., "Tested on Win10 64-bit; needs Linux validation")

## Summary by Sourcery

Add robust audio and settings config load/save behavior and comprehensive tests for audio settings persistence and interaction with other systems.

Bug Fixes:
- Ensure AudioManager.load_volumes validates config types, logs errors appropriately, updates current_config_path consistently, and applies volumes after loading.
- Make Globals._load_settings robust to missing files and invalid types/values for log level and difficulty without overriding current settings.
- Prevent CI Codecov upload steps from failing the workflow when JUnit XML reports are missing by conditionally uploading only if files exist.

Enhancements:
- Improve separation of concerns between audio, input, and global settings so each manager only reads/writes its own config sections while preserving others.
- Tighten error handling and logging for config load/save failures and corrupted or unwritable config files across audio and global settings.

CI:
- Harden gdunit4, GUT, and browser test workflows to skip Codecov uploads gracefully when test report files are absent.

Tests:
- Add extensive GUT test coverage for AudioManager save/load, reset, and error edge cases, including interaction with Globals and Settings and preservation of non-audio config sections.
- Extend existing audio-related tests to respect per-test config paths and ensure they do not interfere with shared configuration.

### CODE REVIEW

File: scripts/audio_manager.gd [hunk 1]
```diff
@@ -147,27 +147,43 @@ func set_muted(bus_name: String, muted: bool) -> void:
...
+					Globals.log_message(
+						"Invalid type for " + muted_key + ": " + str(typeof(loaded_muted)),
+						Globals.LogLevel.WARNING
+					)
+		Globals.log_message("Loaded volumes from config.", Globals.LogLevel.DEBUG)
+	elif err == ERR_FILE_NOT_FOUND:
+		Globals.log_message("No audio config file found, using defaults.", Globals.LogLevel.DEBUG)
+	else:
+		Globals.log_message("Failed to load audio config: " + str(err), Globals.LogLevel.ERROR)
+	apply_all_volumes()
```
[the remark this answers is not in the document]
Comment (ikostan) [reply scripts/audio_manager.gd]: Feedback invalid: apply_all_volumes() was already unconditional in original load_volumes(), applying defaults on failure. No behavioral regression; tests unaffected as double-call is harmless and pre-existing. No code changes needed.
